### ISSUE TITLE
Doesn't work with pages generated with JS

### ISSUE THREAD

**Issue by peterbe** [pocketjoso/penthouse#159]
So many sites these days have a HTML (from the server) that is just 
```
<html>
  <head><link rel="stylesheet" href="..."></head>
  <body>
    <div id="root"></div>
    <script src="angularreactvue.js"></script>
  </body>
</html>
```

To to figure out the critical CSS, you'd load it a little.

**Comment by pocketjoso** [comment]
The goal of using critical css is to make the first render happen without having to make any additional requests after the html, neither for JS nor for extra CSS.

Until your site shows server rendered meaningful html, critical css is of no use.

**Comment by peterbe** [comment]
Arg. I lost my unfinished comment. Anyway, I arrived here to file my issue after testing your (old) online version that is based on penthouse. 
Then I looked up the code and it does seem to use phantomJS to do the downloading but the `blockJSRequests` option is **on** by default. 

I just tested it, with this:
```javascript
penthouse({
  url: 'http://localhost:3000',
  css: path.join(__basedir + 'bootstrap.min.css'),
  // OPTIONAL params
  width: 1300,
  height: 3000,
  forceInclude: [
    '.keepMeEvenIfNotSeenInDom',
    /^\.regexWorksToo/
  ],
  timeout: 30000,
  strict: false,
  maxEmbeddedBase64Length: 1000,
  userAgent: 'Penthouse Critical Path CSS Generator',
  renderWaitTime: 2000,
  blockJSRequests: true,
  phantomJsOptions: {
    // 'proxy': 'http://proxy.company.com:8080',
    // 'ssl-protocol': 'SSLv3'
  },
  customPageHeaders: {
  }
}, function(err, criticalCss) {
  if (err) {
      // handle error
      throw err;
  }
  // fs.writeFileSync('outfile.css', criticalCss);
  console.log(criticalCss)
});
```
(Note the use of `blockJSRequests: true`). Output as follows:
```
▶ node index.js | wc -c
    1067
```
Also, I can see exactly which resources phantom requested because I debug all the request in my express debug server:
```
  express:router dispatching GET / +18s
  express:router query  : / +0ms
  express:router expressInit  : / +0ms
  express:router compression  : / +0ms
  express:router webpackDevMiddleware  : / +0ms
  express:router dispatching GET /static/bootstrap.min.css +4ms
  express:router query  : /static/bootstrap.min.css +0ms
  express:router expressInit  : /static/bootstrap.min.css +0ms
  express:router compression  : /static/bootstrap.min.css +0ms
  express:router webpackDevMiddleware  : /static/bootstrap.min.css +0ms
  express:router dispatching GET /static/addtohomescreen.css +2ms
  express:router query  : /static/addtohomescreen.css +0ms
  express:router expressInit  : /static/addtohomescreen.css +0ms
  express:router compression  : /static/addtohomescreen.css +1ms
  express:router webpackDevMiddleware  : /static/addtohomescreen.css +0ms
  express:router dispatching GET /static/print.css +15ms
  express:router query  : /static/print.css +0ms
  express:router expressInit  : /static/print.css +0ms
  express:router compression  : /static/print.css +0ms
  express:router webpackDevMiddleware  : /static/print.css +0ms
```

However, if I **change** the config to `blockJSRequests: false` and run it again, I get this output:

```
▶ node index.js | wc -c
    6104
```

And the server logs now have .js files in it:
```
  express:router dispatching GET / +2m
  express:router query  : / +0ms
  express:router expressInit  : / +0ms
  express:router compression  : / +1ms
  express:router webpackDevMiddleware  : / +0ms
  express:router dispatching GET /static/bootstrap.min.css +4ms
  express:router query  : /static/bootstrap.min.css +0ms
  express:router expressInit  : /static/bootstrap.min.css +0ms
  express:router compression  : /static/bootstrap.min.css +0ms
  express:router webpackDevMiddleware  : /static/bootstrap.min.css +0ms
  express:router dispatching GET /static/addtohomescreen.css +1ms
  express:router query  : /static/addtohomescreen.css +0ms
  express:router expressInit  : /static/addtohomescreen.css +0ms
  express:router compression  : /static/addtohomescreen.css +1ms
  express:router webpackDevMiddleware  : /static/addtohomescreen.css +0ms
  express:router dispatching GET /static/addtohomescreen.min.js +1ms
  express:router query  : /static/addtohomescreen.min.js +0ms
  express:router expressInit  : /static/addtohomescreen.min.js +0ms
  express:router compression  : /static/addtohomescreen.min.js +1ms
  express:router webpackDevMiddleware  : /static/addtohomescreen.min.js +0ms
  express:router dispatching GET /static/js/bundle.js +0ms
  express:router query  : /static/js/bundle.js +0ms
  express:router expressInit  : /static/js/bundle.js +0ms
  express:router compression  : /static/js/bundle.js +0ms
  express:router webpackDevMiddleware  : /static/js/bundle.js +0ms
  express:router dispatching GET /static/burger.svg +306ms
  express:router query  : /static/burger.svg +0ms
  express:router expressInit  : /static/burger.svg +0ms
  express:router compression  : /static/burger.svg +0ms
  express:router webpackDevMiddleware  : /static/burger.svg +0ms
  express:router dispatching GET /static/offline-sprite.png +1ms
  express:router query  : /static/offline-sprite.png +0ms
  express:router expressInit  : /static/offline-sprite.png +0ms
  express:router compression  : /static/offline-sprite.png +0ms
  express:router webpackDevMiddleware  : /static/offline-sprite.png +0ms
  express:router dispatching GET /static/print.css +37ms
  express:router query  : /static/print.css +0ms
  express:router expressInit  : /static/print.css +0ms
  express:router compression  : /static/print.css +0ms
  express:router webpackDevMiddleware  : /static/print.css +0ms
```

That means it works awesomely! It's able to wait till React has rendered a bunch of stuff such as the welcome page with its nav header etc. (In the second instance `.navbar-nav {` is in the output)

In conclusion, this is awesome. I'm going to try to use this to build a tool that calculates the minimal CSS needed based simply on a list of URLs.

**Comment by pocketjoso** [comment]
@peterbe I hope you understand that this generated critical css will still not be of any use to you for performance reasons, if you will keep using an empty html and relying on JS for rendering. By the time JS has loaded, so will your full css, and hence the critical css serves no purpose.

The first step to improve render performance for a SPA is always to make it server rendered. Only after that does critical css come into play.

**Comment by peterbe** [comment]
Server rendered is not always an option. E.g. you have a react app hosted on Firebase. Or you have some server-side rendered HTML but once the JS is fully loaded, then progressively you render more stuff. For example AJAX'ing comments, related links, a side bar, etc. 

The goal is to give the user something as a soon as possible and build it up once the user can start working with what they've got so far.
